modules/ui/src/main/scala/lucuma/ui/forms/FormInputEV.scala

```diff
@@ -115,7 +109,7 @@ object FormInputEV {
         state.curValue,
         { validated =>
           val validatedCB = validated match {
-            case Valid(a)   => props.valSet(a)
+            case Valid(a)   => props.valSet(a).when(state.prevValue =!= state.curValue)
```

Comment:
I think you might need `props.valGet =!= state.prevValue` because the text in the input (`curValue`) might not be in normal form. I'm actually changing them to state.displayValue && display.modelValue so that is clearer. We could just use `A` instead of string for `modelValue`, but that would require a `Eq` instance for `A`

Reply:
Good catch. Comparing `A`s with an `Eq` is the right thing to do.

I don't think `props.valGet =!= state.prevValue` works since that will probably always be true, in that case we are not comparing with the current input value (`state.prevValue`  gets set to `props.valGet`).

I'm going to experiment with the `Eq` approach and will push another commit.